### ISSUE TITLE
parameters from sails.io.js don't get received by custom controller methods

### ISSUE THREAD

**Issue by barroudjo** [balderdashy/sails#3052]
I have a node server connected with sails.io.js to a sails server.
When I query a blueprint route (not a custom controller method) for example for a model called "themes", i can do:

````
var params =  {where: {subject: 'foo'}, sort: 'bar desc'};
io.socket.get('/themes', params, function (result) {...}
````
and sails properly parses params.

However if I do the same with a custom controller method, say:
````
var params = {slug: 'foo', sort: 'bar desc};
io.socket.get('/themes/getnews', params, function (result) {...}
````
Neither in req.query nor in req.params can I find these params.

The only way I found to send parameters to custom methods in controllers is to put them directly in the query string:
````
io.socket.get('/themes/getnews?slug=foo&sort=bar%20desc', function (result) {...}
````
And then I can find them in req.query. But this is unnecessarily heavy... Can't the first option work ?

Thanks !

**Comment by sailsbot** [comment]
Thanks for posting, @barroudjo.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
